hr_employee_lastnames/models/hr_employee.py

```diff
@@ -0,0 +1,160 @@
+import logging
+
+from odoo import api, models
+
+from odoo.addons.hr_employee_firstname.models.hr_employee import UPDATE_PARTNER_FIELDS
+
+_logger = logging.getLogger(__name__)
+
+UPDATE_PARTNER_FIELDS += ["lastname2"]
+
+
+class HrEmployee(models.Model):
+    _inherit = "hr.employee"
+
+    @api.model
+    def _get_name_lastnames(self, lastname, firstname, lastname2=None):
+        order = self._get_names_order()
+        names = list()
+        if order == "first_last":
+            if firstname:
+                names.append(firstname)
+            if lastname:
+                names.append(lastname)
+            if lastname2:
+                names.append(lastname2)
+        else:
+            if lastname:
+                names.append(lastname)
+            if lastname2:
+                names.append(lastname2)
+            if names and firstname and order == "last_first_comma":
+                names[-1] = names[-1] + ","
+            if firstname:
+                names.append(firstname)
+        return " ".join(names)
+
+    def _prepare_vals_on_create_firstname_lastname(self, vals):
+        values = vals.copy()
+        res = super()._prepare_vals_on_create_firstname_lastname(values)
+        if any([field in vals for field in ("firstname", "lastname", "lastname2")]):
+            vals["name"] = self._get_name_lastnames(
+                vals.get("lastname"), vals.get("firstname"), vals.get("lastname2")
+            )
+        elif vals.get("name"):
+            name_splitted = self.split_name(vals["name"])
+            vals["firstname"] = name_splitted["firstname"]
+            vals["lastname"] = name_splitted["lastname"]
+            vals["lastname2"] = name_splitted["lastname2"]
+        return res
+
+    def _prepare_vals_on_write_firstname_lastname(self, vals):
+        values = vals.copy()
+        res = super()._prepare_vals_on_write_firstname_lastname(values)
+        if any([field in vals for field in ("firstname", "lastname", "lastname2")]):
+            if "lastname" in vals:
+                lastname = vals["lastname"]
+            else:
+                lastname = self.lastname
+            if "firstname" in vals:
+                firstname = vals["firstname"]
+            else:
+                firstname = self.firstname
+            if "lastname2" in vals:
+                lastname2 = vals["lastname2"]
+            else:
+                lastname2 = self.lastname2
+            vals["name"] = self._get_name_lastnames(lastname, firstname, lastname2)
+        elif vals.get("name"):
```

Comment:
Hi @andreagidaltig, i created a PR fixing some issues in the v14 module ( #1283) I could reproduce the bug on the runboat v17. 
To reproduce you can create an employee with a first name equals to " Name 1 Name 2 name 3" and a  lastname equals to "name4"  if you export the data (including the external_id) and importing again the data, the name will be changed. 

Could you fix it or cherry pick my commit? Thanks

Reply:
@FrancoMaxime since your PR is not merged yet, it shouldn't be included in the migration (because it's still subject to changes).

Reply:
@luisg123v Thanks, I hadn’t thought of that!